Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -338,6 +340,56 @@ static uint32_t convertSystemLayoutDirection(NSUserInterfaceLayoutDirection dire
 
 #endif // PLATFORM(MAC)
 
+- (void)_initializeScreenTimeWebpageController
+{
+#if ENABLE(SCREEN_TIME)
+    if (!_page->preferences().screenTimeEnabled())
+        return;
+    if (!_screenTimeWebpageController) {
+        _screenTimeWebpageController = adoptNS([[STWebpageController alloc] init]);
+        [_screenTimeWebpageController addObserver:self forKeyPath:@"URLIsBlocked" options:(NSKeyValueObservingOptionOld | NSKeyValueObservingOptionNew) context:&readyForDisplayKVOContext];
+#if PLATFORM(MAC)
+        NSView *screenTimeView = [_screenTimeWebpageController view];
+        screenTimeView.frame = self.bounds;
```

Comment:
Who keeps the frame up to date if the web view resizes?

Reply:
Yeah, we probably want to add some logic in here:

```
void WebViewImpl::setFrameSize(CGSize)
{
    [m_layoutStrategy didChangeFrameSize];

    auto updatedBounds = [m_view bounds];
    [m_warningView setFrame:updatedBounds];
    [m_page->screenTimeWebpageController() view].frame = updatedBounds;
}
```